test/generators/typescript/TypeScriptGenerator.spec.ts

```diff
@@ -488,4 +488,200 @@ ${content}`;
       expect(discriminatorEnum?.result).toMatchSnapshot();
     });
   });
+
+  describe('Combine oneOf and allOf', () => {
+    const asyncapiDoc = {
+      asyncapi: '2.4.0',
+      info: {
+        title: 'Pet',
+        version: '1.0.0'
+      },
+      channels: {},
+      components: {
+        messages: {
+          Pet: {
+            payload: {
+              title: 'Pet',
+              allOf: [
+                { $ref: '#/components/schemas/Animal' },
+              ],
+              oneOf: [
+                { $ref: '#/components/schemas/Cat' },
+                { $ref: '#/components/schemas/Dog' },
+              ],
```

Comment:
```json
{
              allOf: [
                { $ref: '#/components/schemas/Animal' },
                {oneOf: [
                  { $ref: '#/components/schemas/Cat' },
                  { $ref: '#/components/schemas/Dog' },
                ]}
              ],
}
```

Is this a pattern you have seen being used?

Reply:
No, just the one in my test.